src/modules/expression-evaluator/src/evaluateExpression.ts

```diff
@@ -290,9 +287,21 @@ const graphQLquery = async (
   return data.data
 }
 
-// GET request using fetch (node or browser variety)
-const fetchAPIdata = async (url: string, APIfetch: any) => {
-  const result = await APIfetch(url)
-  const data = await result.json()
-  return data
+interface APIrequestProps {
+  url: string
+  APIfetch: any
+  method?: 'GET' | 'POST'
+  body?: { [key: string]: string } | null
+}
+
+// GET/POST request using fetch (node or browser variety)
+const fetchAPIrequest = async ({ url, APIfetch, method = 'GET', body }: APIrequestProps) => {
+  const result = await APIfetch(url, {
+    method,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify(body),
```

Comment:
Although this work, would prefer a condition:

![Screen Shot 2021-05-13 at 11 23 16 AM](https://user-images.githubusercontent.com/26194949/118056496-292f2800-b3de-11eb-8ff0-cb53e0a496e5.png)

```
body: body instanceof Object ?  JSON.stringify(body) : undefined
```

Reply:
Not sure what the difference is, since `JSON.stringify(body)` returns undefined anyway, if body is undefined. And any other primitive type is valid JSON.

Reply:
I was worried about error being thrown (if body is not valid), but it's all wrapped with error handlers, so should be all good